Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/ManipulationHandler.cs

```diff
@@ -501,80 +491,76 @@ private void HandleTwoHandManipulationUpdated()
-                RotateInOneHandType rotateInOneHandType = isNearManipulation ? oneHandRotationModeNear : oneHandRotationModeFar;
-                if (rotateInOneHandType == RotateInOneHandType.MaintainOriginalRotation)
-                {
-                    targetRotation = hostTransform.rotation;
-                }
-                else if (rotateInOneHandType == RotateInOneHandType.MaintainRotationToUser)
-                {
-                    targetRotation = CameraCache.Main.transform.rotation * startObjectRotationCameraSpace;
-                }
-                else if (rotateInOneHandType == RotateInOneHandType.GravityAlignedMaintainRotationToUser)
-                {
-                    var cameraForwardFlat = CameraCache.Main.transform.forward;
-                    cameraForwardFlat.y = 0;
-                    targetRotation = Quaternion.LookRotation(cameraForwardFlat, Vector3.up) * startObjectRotationFlatCameraSpace;
-                }
-                else if (rotateInOneHandType == RotateInOneHandType.FaceUser)
-                {
-                    Vector3 directionToTarget = hostTransform.position - CameraCache.Main.transform.position;
-                    targetRotation = Quaternion.LookRotation(-directionToTarget);
-                }
-                else if (rotateInOneHandType == RotateInOneHandType.FaceAwayFromUser)
-                {
-                    Vector3 directionToTarget = hostTransform.position - CameraCache.Main.transform.position;
-                    targetRotation = Quaternion.LookRotation(directionToTarget);
-                }
-                else
+                targetRotation = hostTransform.rotation;
+            }
+            else if (rotateInOneHandType == RotateInOneHandType.MaintainRotationToUser)
+            {
+                targetRotation = CameraCache.Main.transform.rotation * startObjectRotationCameraSpace;
+            }
+            else if (rotateInOneHandType == RotateInOneHandType.GravityAlignedMaintainRotationToUser)
+            {
+                var cameraForwardFlat = CameraCache.Main.transform.forward;
+                cameraForwardFlat.y = 0;
+                targetRotation = Quaternion.LookRotation(cameraForwardFlat, Vector3.up) * startObjectRotationFlatCameraSpace;
+            }
+            else if (rotateInOneHandType == RotateInOneHandType.FaceUser)
+            {
+                Vector3 directionToTarget = hostTransform.position - CameraCache.Main.transform.position;
+                targetRotation = Quaternion.LookRotation(-directionToTarget);
+            }
+            else if (rotateInOneHandType == RotateInOneHandType.FaceAwayFromUser)
+            {
+                Vector3 directionToTarget = hostTransform.position - CameraCache.Main.transform.position;
+                targetRotation = Quaternion.LookRotation(directionToTarget);
+            }
+            else
+            {
+                targetRotation = pointer.Rotation * objectToHandRotation;
```

Comment:
Don't you also need to update objectToHandRotation to use the pointer pose?

Reply:
It seems that is was already using it:

            // Calculate relative transform from object to hand.
            Quaternion worldToPalmRotation = Quaternion.Inverse(pointer.Rotation);
            objectToHandRotation = worldToPalmRotation * hostTransform.rotation;